src/Ide/Arguments.hs

```diff
@@ -138,12 +140,40 @@ arguments plugins = GhcideArguments
           <> help "Sets the log level to Debug, alias for '--log-level Debug'"
           )
         )
-      <*> optional (strOption
-           (long "logfile"
-          <> short 'l'
```

Comment:
I would suggest keeping the short form for at least one release cycle.
I just got bitten by this, and had to re-configure my emacs to use it 

```elisp
  :custom
  (lsp-haskell-server-args `("-d" "--log-file" ,lsp-haskell-server-log-file))
```

Reply:
Ah sorry, that was an oversight, I meant to leave the old options in! I left the old spelling but missed the short option.

Reply:
Although I don't think leaving it in for a release cycle helps all that much: it just means that people will be surprised by it one cycle later. It's only useful if you're switching between and old and new version.

Reply:
(... then why am I leaving the old spelling in? I guess we could just never remove it)

Reply:
The one release cycle thing is so you could update clients to use the new one, but if people still ran an older server it would still work. Then in time get rid of it.
To me, keeping the short form is the best option

Reply:
https://github.com/haskell/haskell-language-server/pull/3672